pkg/engine2/operational_eval/eval.go

```diff
@@ -230,8 +226,18 @@ func (eval *Evaluator) cleanupPropertiesSubVertices(ref construct.PropertyRef, v
 			continue
 		}
 		if strings.HasPrefix(key.Ref.Property, ref.Property) {
-			err = eval.unevaluated.RemoveVertex(key)
-			errs = errors.Join(errs, err)
+
+			path, err := resource.PropertyPath(key.Ref.Property)
+			// an error would mean that the path no longer exists so we know we should remove the vertex
+			if err == nil {
+				// if the paths parent still exists then we know we will end up evaluating the vertex and should not remove it
+				parentIndex := len(path) - 2
+				if parentIndex < 0 || path[parentIndex].Get() != nil {
+					continue
+				}
+			}
+			errs = errors.Join(errs, graph_addons.RemoveVertexAndEdges(eval.graph, key))
+			errs = errors.Join(errs, graph_addons.RemoveVertexAndEdges(eval.unevaluated, key))
```

Comment:
```suggestion
			errs = errors.Join(errs,
				graph_addons.RemoveVertexAndEdges(eval.graph, key),
				graph_addons.RemoveVertexAndEdges(eval.unevaluated, key),
			)
```
nit: a little cleaner